interview-questions.md

```diff
@@ -9,43 +9,60 @@ Then, research the question to expand on your answer. Even if you feel you have
 1. What is a hash in Ruby?
 
 Your answer:
+Inside of Ruby a Hash is a collection of key: value pairs, in order to create a Hash we can use curly braces or the keyword Hash.new. For the first method we would create a variable, I will call it hash_method_one. Now I will assign that variable to a set of curly braces, and inside that I would then create my key:value pairs. I could have a key named name, followed by a colon then my value of Luis inside single quotes. This would be my Hash. The next method I cold use is the keyword Hash.new. In order to use this method I need to also create a variable. I will go with a variable name of has_method_two. Now I will assign this variable to Hash.new. I have created my Hash and so now I need to add a key:value to it. I would call my variable and use bracket notation to pass my KEY using a colon. Then outside I would assign this to my the value I want. These are two ways of creating a Hash with key:values, but there are some more steps to reassign those key:values or delete them. Changing the value would look the same as the original assigning of the value but now you would change it to whatever you would want. Deleting it would involve using the .delete on the variable. The .delete needs parenthesis so here you would pass the key you want to delete. These are the two ways to create a Hash, I personally prefer using the second newer method.
 
-Researched answer:
+<!-- Trying to verbalize making it and also making examples if I were to be asked to code this  -->
+
+Example for method 1
+
+hash_method_one = {name:'Luis'}
+
+
+Example for method 2 
+
+hash_method_two = Hash.new
+has_method_two[:name] = 'Luis' <---- Created a key:value of name:'Luis'
+has_method_two[:name] = 'Reassigned' <--- The value 'Luis' is now 'Reassigned'
+has_method_two.delete(:name) <--- I deleted my key:value here
+
+Researched answer: 
+I completely forgot to add the usage of the Hash rocket, this is how we create key:value pair with a key being a different data type. This involves the usage of the first method using curly braces. If I wanted my key to be a string, I would put it in single quotes then use Hash rocket =>. My value would follow after the Hash rocket. Now I can have a string key and string value. Upon some research I learned its also referred to as a fat comma. Also I will have to say that the key is a Symbol.
```

Comment:
Good and close on the symbol! Fat commas were used before symbols were around. now symbols take the lace of those strings and add some special characteristics to the datatype.